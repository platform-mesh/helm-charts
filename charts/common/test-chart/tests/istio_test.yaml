suite: istio
templates:
  - vs.yaml
values:
  - ../values.yaml
chart:
  namespace: default
tests:
  - it: test global value
    asserts:
      - matchSnapshot: {}
    set:
      global.istio.enabled: true
  - it: test local value
    asserts:
      - matchSnapshot: {}
    set:
      istio.enabled: true
  - it: test local value disabled
    asserts:
      - matchSnapshot: {}
    set:
      istio.enabled: false
  - it: test test no value
    asserts:
      - matchSnapshot: {}
