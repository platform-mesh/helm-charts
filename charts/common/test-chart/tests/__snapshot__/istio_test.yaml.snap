test global value:
  1: |
    apiVersion: networking.istio.io/v1beta1
    kind: VirtualService
    metadata:
      name: RELEASE-NAME-test-chart
    spec:
      gateways:
        - NAMESPACE/gateway
      hosts:
        - some.domain
      http:
        match:
          - uri:
              exact: some.domain
        name: default
        route:
          - destination:
              host: some.host
              port:
                number: 8080
test local value:
  1: |
    apiVersion: networking.istio.io/v1beta1
    kind: VirtualService
    metadata:
      name: RELEASE-NAME-test-chart
    spec:
      gateways:
        - NAMESPACE/gateway
      hosts:
        - some.domain
      http:
        match:
          - uri:
              exact: some.domain
        name: default
        route:
          - destination:
              host: some.host
              port:
                number: 8080
test test no value:
  1: |
    apiVersion: networking.istio.io/v1beta1
    kind: VirtualService
    metadata:
      name: RELEASE-NAME-test-chart
    spec:
      gateways:
        - NAMESPACE/gateway
      hosts:
        - some.domain
      http:
        match:
          - uri:
              exact: some.domain
        name: default
        route:
          - destination:
              host: some.host
              port:
                number: 8080
